arrow/src/main/java/org/apache/iceberg/arrow/vectorized/parquet/VectorizedPageIterator.java

```diff
@@ -95,7 +94,9 @@ protected void initDataReader(Encoding dataEncoding, ByteBufferInputStream in, i
         throw new ParquetDecodingException("could not read page in col " + desc, e);
       }
     } else {
-      if (dataEncoding != Encoding.PLAIN) {
+      if (dataEncoding == Encoding.PLAIN) {
+        valuesReader = new VectorizedPlainValuesReader();
+      } else {
```

Comment:
More VectorizedValuesReader implementations will be added here for different encodings.